internal/service/cache/cache.go

```diff
@@ -0,0 +1,118 @@
+package cache
+
+import (
+	"time"
+
+	"github.com/grafana/dskit/cache"
+	"github.com/grafana/dskit/flagext"
+	"github.com/pkg/errors"
+)
+
+const (
+	// InMemory is the value for the in-memory cache backend.
+	InMemory = "inmemory"
+
+	// Memcached is the value for the Memcached cache backend.
+	Memcached = cache.BackendMemcached
+
+	// Redis is the value for the Redis cache backend.
+	Redis = cache.BackendRedis
+
+	// Default is the value for the default cache backend.
+	Default = InMemory
+)
+
+var (
+	SupportedCaches = []string{InMemory, Memcached, Redis}
+
+	errUnsupportedCache = errors.New("unsupported cache backend")
+	errNotFound         = errors.New("not found in cache")
+)
+
+type CacheConfig struct {
+	Backend   string              `alloy:"backend,attr"`
+	Memcached MemcachedConfig     `alloy:"memcached,block,optional"`
+	Redis     RedisConf           `alloy:"redis,block,optional"`
+	InMemory  InMemoryCacheConfig `alloy:"inmemory,block,optional"`
+}
+
+//TODO Those field are copied from dskit/cache for now (only the one mandatory)
+// We need to have a better way to manage conf
+// For now I used those because we cannot embed 'yaml' tags into alloy configuration
+// Ideally we should be using the dskit/cache conf directly, but it means it should not
+// be into the alloy configuration ?
+
+type RedisConf struct {
```

Comment:
As explained in the comment, I'm open to suggestion here on how to handle the config part.

For now each cache is configured at the relabel component level but this include copying the struct to add the `alloy` tags as we cannot embed the dskit/cache configs into the grafana alloy config.

If we decide to move this config outside of the alloy config we could directly use the dskit/cache config.